packages/devextreme/js/__internal/ui/splitter/utils/layout.ts

```diff
@@ -123,14 +124,106 @@ export function updateItemsSize(items, sizeDistribution): void {
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+  return isNumeric(size) || (isString(size) && size.endsWith('px'));
+}
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+function calculatePercentage(totalSize, size) {
+  if (totalSize === 0) {
+    return 0;
+  }
+
+  const percentage = (size / totalSize) * 100;
+  return percentage;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+function getPercentSize(size: string | number, totalPanesSize: number): number {
+  const isPixel = isPixelWidth(size);
+  const sizeNumber = parseFloat(size as string);
+
+  if (isPixel) {
+    return calculatePercentage(totalPanesSize, sizeNumber);
+  }
+
+  const isPercentage = isPercentWidth(size);
+  if (isPercentage) {
+    return sizeNumber;
+  }
+
+  // todo: handle incorrect size input
+  return 0;
+}
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export function getInitialLayout(panes, totalPanesSize: number): number[] {
   const layout: number[] = [];
+  let totalSize = 0;
+  let sizeOverflow = false;
+
+  // eslint-disable-next-line no-restricted-syntax
+  for (const pane of panes) {
+    if (pane.visible === false || sizeOverflow || pane.size === 0) {
+      layout.push(0);
+      // todo: refactor
+    } else if (pane.size && (isPercentWidth(pane.size) || isPixelWidth(pane.size))) {
+      let percentSize = getPercentSize(pane.size, totalPanesSize);
+      percentSize = Math.min(100 - totalSize, percentSize);
+      totalSize += percentSize;
+      layout.push(percentSize);
+
+      if (totalSize >= 100) {
+        sizeOverflow = true;
+      }
+    } else {
+      layout.push(-1);
+    }
+  }
 
-  const visibleItemsCount = items.filter((item) => item.visible !== false).length;
+  const noSizePanes = panes.filter((p) => p.visible !== false && !p.size && p.size !== 0);
 
-  items.forEach((item) => {
-    layout.push(item.visible === false ? 0 : 100 / visibleItemsCount);
-  });
+  if (noSizePanes.length) {
+    const remainingSpace = Math.max(100 - totalSize, 0);
+
+    layout.forEach((pane, index) => {
+      if (layout[index] === -1) {
+        layout[index] = remainingSpace / noSizePanes.length;
+      }
+    });
+  } else if (totalSize < 100) {
+    layout[findLastIndexOfVisibleItem(panes)] += 100 - totalSize;
+  }
 
   return layout;
 }
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+function getElementItemsSizeSum($element, orientation, handlesCount): number {
+  const splitterSize = $element.get(0).getBoundingClientRect();
+  const size: number = orientation === ORIENTATION.horizontal
+    ? splitterSize.width : splitterSize.height;
+
+  const handlesSizeSum = handlesCount * DEFAULT_RESIZE_HANDLE_SIZE;
+
+  return size - handlesSizeSum;
+}
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export function getElementSize($element, items, orientation, width, height): number {
```

Comment:
"If Pane.size is set to '10%', it should represent 10% of the Splitter's total width. It seems that in your case, it's 10% of the total width of the panes, excluding the width of the resize handles. Please check this and decide what way is right.